Tests/ViewInspectorTests/SwiftUI/CustomViewTests.swift

```diff
@@ -50,12 +50,12 @@ final class CustomViewTests: XCTestCase {
         XCTAssertNoThrow(try view.inspect().view(EnvironmentStateTestView.self))
     }
 
-    func testEnvironmentObjectActualView() throws {
-        let viewModel = ExternalState()
-        let view = EnvironmentStateTestView().environmentObject(viewModel)
-        _ = try view.inspect().view(EnvironmentStateTestView.self)
-            .actualView().viewModel
-    }
+//    func testEnvironmentObjectActualView() throws {
```

Comment:
Commit b5364cf3f93c834cc96ded580ae8267deb7784ec resulted in this test crashes.
<img width="2119" alt="Screen Shot 2022-11-28 at 3 59 15 PM" src="https://user-images.githubusercontent.com/1791049/204405703-4608d8a8-4755-4437-a245-cdba45bf0d19.png">


The same commit also resulted in many other tests starting to fail.
<img width="1089" alt="Screen Shot 2022-11-28 at 3 57 33 PM" src="https://user-images.githubusercontent.com/1791049/204405648-e39d34fe-03bc-4890-9251-d6b56749dc24.png">

Reply:
My goal in this commit is to refactor the environment injections functions to be static functions that operate on a view rather than functions that are called directly on a view. This change is necessary to relax the restriction that all views conform to `Inspectable`.

Reply:
@nalexn could you advise me on the WIP changes to EnvironmentInjection.swift? I can't see how I've functionally changed those methods though it's clear from the crashing/failing tests that I've done something to alter the behavior. I'm hoping the bug may be more obvious to someone who has spent more time in this code. Thanks!

Reply:
I wonder if the issue with my refactor may be related to structs being copied.

Reply:
Thanks @erichoracek for helping me resolve the crash in `testEnvironmentObjectActualView()` and many of the failing tests. Eric discovered that I needed to update [these lines](https://github.com/nalexn/ViewInspector/blob/b65b8381e699a4ae3ae82f5806ac5e40b3676b2e/Sources/ViewInspector/EnvironmentInjection.swift#L30-L32) to be `MemoryLayout<T>` instead of `MemoryLayout<Self>`.

Reply:
I now see that b65b8381e699a4ae3ae82f5806ac5e40b3676b2e is still causing some tests to fail. I will need to investigate these next.

Test failures before applying b65b8381e699a4ae3ae82f5806ac5e40b3676b2e. It's expected that this test fails since I broke the test intentionally in the first commit.

<img width="488" alt="Screen Shot 2022-12-01 at 2 16 09 PM" src="https://user-images.githubusercontent.com/1791049/205171625-a92736e0-ea3a-46ea-9d41-a471095137ba.png">

Test failures after applying b65b8381e699a4ae3ae82f5806ac5e40b3676b2e.
<img width="974" alt="Screen Shot 2022-12-01 at 2 17 44 PM" src="https://user-images.githubusercontent.com/1791049/205171675-da8c81c2-ded1-47c6-84d0-f571a35589ec.png">